test/integration/framework/perf_utils.go

```diff
@@ -89,7 +90,20 @@ func (p *IntegrationTestNodePreparer) PrepareNodes(ctx context.Context, nextNode
 	for i := 0; i < numNodes; i++ {
 		var err error
 		for retry := 0; retry < retries; retry++ {
-			_, err = p.client.CoreV1().Nodes().Create(ctx, baseNode, metav1.CreateOptions{})
+			// Create nodes with the usual kubernetes.io/hostname label.
+			// For that we need to know the name in advance, if we want to
+			// do it in one request.
+			node := baseNode.DeepCopy()
+			name := node.Name
+			if name == "" {
+				name = node.GenerateName + rand.String(8)
```

Comment:
nit:

```suggestion
				name = node.GenerateName + rand.String(5)
```

to match the apiserver behavior